tools/check_release.py

```diff
@@ -0,0 +1,455 @@
+    log_message("Task accepted and given ID: %s" % uuid)
+    sys.stdout.write("\t\tCompiling: ")
+    result = False
+    fail_type = None
+
+    # poll for output
+    for check in range(0, polls):
+        sys.stdout.write('.')
+        time.sleep(1)
+        r = requests.get(url + "output/%s" % uuid, auth=auth)
+        response = r.json()
+        if response['result']['data']['task_complete']:
+            
+            # Task completed. Now determine the result. Should be one of :
+            # 1) Successful compilation
+            # 2) Failed compilation with an error message
+            # 3) Internal failure of the online compiler            
+            result = bool(response['result']['data']['compilation_success'])
+            if result:
+                sys.stdout.write("SUCCESSFUL \n")
+            else:
+                # Did this fail due to a genuine compilation error or a failue of the api itself ?
+                sys.stdout.write("FAILURE \n")
+                sys.stdout.flush()
+                fail_type = get_compilation_failure(response['result']['data']['new_messages'])
+            break
+    if not result and fail_type == None:
+        fail_type = "Internal"
+        
+    return result, fail_type
+
+
+def build_repo(target, program, user, pw, polls=25, url="https://developer.mbed.org/api/v2/tasks/compiler/"):
+    """ Wrapper for sending an API command request to the online IDE. Sends a build request.
+
+    Args:
+    target - Target to be built
+    program - Test program to build
+    user - mbed username
+    pw - mbed password
+    polls - Number of times to poll for results
+    url - THe URL for the online compiler API
+    
+    Returns:
+    result - True/False indicating the success/failure of the compilation
+    fail_type - the failure text if the compilation failed, else None
+    """
+    payload = {'clean':True, 'target':target, 'program':program}
+    auth = (user, pw)
+    return invoke_api(payload, url, auth, polls)
+
+def run_cmd(command, print_warning_on_fail=True, exit_on_failure=False):
+    """ Passes a command to the system and returns a True/False result once the 
+        command has been executed, indicating success/failure. Commands are passed
+        as a list of tokens. 
+        E.g. The command 'git remote -v' would be passed in as ['git', 'remote', '-v']
+
+    Args:
+    command - system command as a list of tokens
+    print_warning_on_fail - If True print any failure warning to the screen
+                            (default = True)
+    exit_on_failure - If True exit the program on failure (default = False)
+    
+    Returns:
+    result - True/False indicating the success/failure of the command
+    """
+    print('[Exec] %s' % ' '.join(command))
+    return_code = subprocess.call(command, shell=True)
+    
+    if return_code:
+        if print_warning_on_fail:
+            print("The command '%s' failed with return code: %s" % (' '.join(command), return_code))
```

Comment:
Can't we just use logging here too? and drop this poorly-named argument `print_warning_on_fail`